File: pyg-seller-web/src/main/webapp/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){
	
	//**************************************************************    	
	// //模板的下拉列表数据
	// this.getBrandOption =function(id){
	// 	return $http.get("../goods/getBrandOption.do?id="+id);
	// }
	//**************************************************************
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../goods/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../goods/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../goods/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../goods/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../goods/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../goods/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../goods/search.do?page='+page+"&rows="+rows, searchEntity);
	}
	//上架
	this.upShelf=function (ids) {
		return $http.get('../goods/upShelf.do?ids='+ids);
    }
    //下架
    this.downShelf=function (ids) {
        return $http.get('../goods/downShelf.do?ids='+ids);
    }
});

File: pyg-sellergoods-interface/src/main/java/com/pyg/sellergoods/service/GoodsService.java
package com.pyg.sellergoods.service;

import java.util.List;

import com.pyg.entity.PageResult;
import com.pyg.pojo.TbGoods;
import com.pyg.pojo.TbItem;
import com.pyg.pojogroup.Goods;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbGoods> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加  修改后的 , 这里需要的是一个组合类
	*/
	public void add(Goods goods);
	
	
	/**
	 * 修改 
	 */
	public void update(Goods goods);
	
	public void updateStatus(Long ids[],String status);
	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Goods findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbGoods goods, int pageNum,int pageSize);

	//查询item信信息导入solr
	List<TbItem> findItemListByGoodsIdandStatus(Long[] goodsIds, String status );

    void upShelf(Long[] ids);

	void downShelf(Long[] ids);
}
